internal/server/handler/middleware.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"github.com/Kreg101/metrics/internal/server/logger"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// logging логирует запрос и ответ посредством middleware
+func logging(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// При вызове logger.Default() возращается единый на весь сервер логгер
+		log := logger.Default()
+
+		start := time.Now()
+
+		responseData := &responseData{}
+		lw := loggingResponseWriter{
+			ResponseWriter: w,
+			responseData:   responseData,
+		}
+
+		// Подменяет w, на свой с логированием
+		h.ServeHTTP(&lw, r)
+
+		duration := time.Since(start).Milliseconds()
```

Comment:
[Feedback]
В go очень круто придумали, что ввели специальный тип time.Duration, у которого много мелких фишек
И одна из них - это очень приятное строковое представление, т.е. можно не беспокоиться и не переводить самому в милисекунды, секунды, минуты и тп - форматтер этого типа сам позаботиться об этом (например, если количество милисекунд слишком вилеко - он представить их часть в секундах, а остальное в милисекундах - намного удобнее читается в логах)

Поэтому я бы оставил `duration := time.Since(start)`, а в логи бы передавал `duration.String()`